### PR TITLE
feat: fungible trait implementation for currencies 

### PR DESCRIPTION
**Pull request by enthusiastmartin**
Adds implementations of fungible traits for currencies adapter.

### CODE REVIEW

File: currencies/src/fungibles.rs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    <T::NativeCurrency as fungible::Inspect<T::AccountId>>::Balance: Into<BalanceOf<T>> + From<BalanceOf<T>>,
+    WithdrawConsequence<BalanceOf<T>>:
+        From<WithdrawConsequence<<T::NativeCurrency as fungible::Inspect<T::AccountId>>::Balance>>,
+    Result<Self::Balance, DispatchError>:
+        From<Result<<T::NativeCurrency as fungible::Inspect<T::AccountId>>::Balance, DispatchError>>,
+    Result<Self::Balance, DispatchError>:
+        From<Result<<T::MultiCurrency as fungibles::Inspect<T::AccountId>>::Balance, DispatchError>>,
+{
+    fn mint_into(asset: Self::AssetId, who: &T::AccountId, amount: Self::Balance) -> DispatchResult {
+        <Pallet<T>>::deposit(asset, who, amount)
```
Comment (Roznovjak) [inline currencies/src/fungibles.rs:91]: why not `mint_into` methods?

Comment (enthusiastmartin) [reply currencies/src/fungibles.rs:91]: it is actually the same. but it can be changed to use mint into, so it is clear.

**Review by Roznovjak — APPROVED** [review]

File: currencies/src/tests_fungibles.rs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                true
+            ));
+            assert_eq!(FungibleCurrencies::<Runtime>::balance(NATIVE_CURRENCY_ID, &ALICE), 90);
+            assert_eq!(FungibleCurrencies::<Runtime>::balance(NATIVE_CURRENCY_ID, &BOB), 10);
+
+            assert_ok!(FungibleCurrencies::<Runtime>::transfer(
+                X_TOKEN_ID, &BOB, &ALICE, 10, true
+            ));
+            assert_eq!(FungibleCurrencies::<Runtime>::balance(X_TOKEN_ID, &BOB), 90);
+            assert_eq!(FungibleCurrencies::<Runtime>::balance(X_TOKEN_ID, &ALICE), 10);
```
Comment (enthusiastmartin) [inline currencies/src/tests_fungibles.rs:80]: can you actually use directly pallet balances and orml tokens storage to verify that it was correctly updated after transfer? 

similar to what it is done in the other tests.

Comment (enthusiastmartin) [reply currencies/src/tests_fungibles.rs:80]: also i would add test with keep alive false, and ensur that it works too.

**Pull request merged**